src/main/java/org/sonarsource/sonarlint/ls/DiagnosticPublisher.java

```diff
@@ -70,11 +70,13 @@ public void initialize(boolean firstSecretDetected) {
     this.firstSecretIssueDetected = firstSecretDetected;
   }
 
-  public void publishDiagnostics(URI f) {
+  public void publishDiagnostics(URI f, boolean onlyHotspots) {
     if(openNotebooksCache.isNotebook(f)) {
       return;
     }
-    client.publishDiagnostics(createPublishDiagnosticsParams(f));
+    if (!onlyHotspots) {
```

Comment:
Great. This may reduce the load in case of many hotspots in a full project scan.